src/context.rs

```diff
@@ -37,70 +37,80 @@
 //! For most commmon uses (one device, one OS thread) it should suffice to create a single context:
 //!
 //! ```
-//! use rustacuda;
 //! use rustacuda::device::Device;
 //! use rustacuda::context::{Context, ContextFlags};
+//! use std::error::Error;
+//! # fn main () -> Result<(), Box<dyn Error>> {
 //!
-//! rustacuda::init(rustacuda::CudaFlags::empty()).unwrap();
-//! let device = Device::get_device(0).unwrap();
-//! let context = Context::create_and_push(ContextFlags::MAP_HOST | ContextFlags::SCHED_AUTO, device).unwrap();
+//! rustacuda::init(rustacuda::CudaFlags::empty())?;
+//! let device = Device::get_device(0)?;
+//! let context = Context::create_and_push(ContextFlags::MAP_HOST | ContextFlags::SCHED_AUTO, device)?;
 //! // call RustaCUDA functions which use the context
 //!
 //! // The context will be destroyed when dropped or it falls out of scope.
 //! drop(context);
+//! # Ok(())
+//! # }
 //! ```
 //!
 //! If you have multiple OS threads that each submit work to the same device, you can get a handle
 //! to the single context and pass it to each thread.
 //!
 //! ```
-//! # use rustacuda;
+//! # use rustacuda::context::{Context, ContextFlags, CurrentContext};
 //! # use rustacuda::device::Device;
-//! # use rustacuda::context::{Context, CurrentContext, ContextFlags};
-//! #
-//! # rustacuda::init(rustacuda::CudaFlags::empty()).unwrap();
-//! # let device = Device::get_device(0).unwrap();
-//! // As before
-//! let context = Context::create_and_push(ContextFlags::MAP_HOST | ContextFlags::SCHED_AUTO, device).unwrap();
+//! # use std::error::Error;
+//! # fn main() -> Result<(), Box<dyn Error>> {
+//! # rustacuda::init(rustacuda::CudaFlags::empty())?;
+//! # let device = Device::get_device(0)?;
+//!   // As before
+//!   let context =
+//!     Context::create_and_push(ContextFlags::MAP_HOST | ContextFlags::SCHED_AUTO, device)?;
+//!   let mut join_handles = vec![];
 //!
-//! let mut join_handles = vec![];
-//! for _ in 0..4 {
+//!   for _ in 0..4 {
 //!     let unowned = context.get_unowned();
 //!     let join_handle = std::thread::spawn(move || {
-//!         CurrentContext::set_current(&unowned).unwrap();

```

Comment:
Did you mean to change the indentation here?

Reply:
Nope, my mistake, I assumed `cargo fmt` was cleaning up my code... apparently not?